src/rabbit_recovery_terms.erl

```diff
@@ -194,11 +194,29 @@ code_change(_OldVsn, State, _Extra) ->
 %%----------------------------------------------------------------------------
 
 open_table(VHost) ->
+    open_table(VHost, false).
+open_table(VHost, HasFailed) ->
     VHostDir = rabbit_vhost:msg_store_dir_path(VHost),
     File = filename:join(VHostDir, "recovery.dets"),
-    {ok, _} = dets:open_file(VHost, [{file,      File},
-                                     {ram_file,  true},
-                                     {auto_save, infinity}]).
+    try
+        {ok, _} = dets:open_file(VHost, [{file,      File},
+                                        {ram_file,  true},
+                                        {auto_save, infinity}])
+    catch _:_ ->
+        case HasFailed of 
+            false ->
+                rabbit_log:warning("open '~p' failed, delete it and try again.", [File]),
+                rabbit_misc:delete_file(File),
```

Comment:
This function does not exist, resulting in clients inability to connect due to the following exception:

```
2019-12-26 07:03:52.511 [error] <0.435.0> CRASH REPORT Process <0.435.0> with 0 neighbours crashed with reason: call to undefined function rabbit_misc:delete_file("/var/folders/gp/53t98z011678vk9rkcb_s6ph0000gn/T/rabbitmq-test-instances/rabbit/mnesia/rabbit/m...")
2019-12-26 07:03:52.512 [error] <0.434.0> Unable to recover vhost <<"/">> data. Reason {badmatch,{error,{{undef,[{rabbit_misc,delete_file,["/var/folders/gp/53t98z011678vk9rkcb_s6ph0000gn/T/rabbitmq-test-instances/rabbit/mnesia/rabbit/msg_stores/vhosts/628WB79CIFDYO9LJI6DKMI09L/recovery.dets"],[]},{rabbit_recovery_terms,open_table,2,[{file,"src/rabbit_recovery_terms.erl"},{line,209}]},{rabbit_recovery_terms,init,1,[{file,"src/rabbit_recovery_terms.erl"},{line,179}]},{gen_server,init_it,2,[{file,"gen_server.erl"},{line,374}]},{gen_server,init_it,6,[{file,"gen_server.erl"},{line,342}]},{proc_lib,init_p_do_apply,3,[{file,"proc_lib.erl"},{line,249}]}]},{child,undefined,rabbit_recovery_terms,{rabbit_recovery_terms,start_link,[<<"/">>]},transient,30000,worker,[rabbit_recovery_terms]}}}}
 Stacktrace [{rabbit_recovery_terms,start,1,[{file,"src/rabbit_recovery_terms.erl"},{line,47}]},{rabbit_queue_index,start,2,[{file,"src/rabbit_queue_index.erl"},{line,516}]},{rabbit_variable_queue,start,2,[{file,"src/rabbit_variable_queue.erl"},{line,480}]},{rabbit_priority_queue,start,2,[{file,"src/rabbit_priority_queue.erl"},{line,94}]},{rabbit_amqqueue,recover_classic_queues,2,[{file,"src/rabbit_amqqueue.erl"},{line,136}]},{rabbit_amqqueue,recover,1,[{file,"src/rabbit_amqqueue.erl"},{line,127}]},{rabbit_vhost,recover,1,[{file,"src/rabbit_vhost.erl"},{line,60}]},{rabbit_vhost_process,init,1,[{file,"src/rabbit_vhost_process.erl"},{line,56}]}]
```